roles/logstash/tasks/logstash-security.yml

```
@@ -427,94 +389,81 @@
     - configuration
     - logstash_configuration
 
-- name: Check for logstash_writer role # noqa: risky-shell-pipe
-  ansible.builtin.shell: >
-    if test -n "$(ps -p $$ | grep bash)"; then set -o pipefail; fi;
-    curl --cacert {{ elasticstack_ca_dir }}/ca.crt
-    -u elastic:{{ logstash_elasticstack_password.stdout }}
-    https://{{ hostvars[groups['elasticsearch'][0]].ansible_default_ipv4.address }}:{{ elasticstack_elasticsearch_http_port }}/_security/role/logstash_writer
-    | grep "enabled"
-  delegate_to: "{{ elasticstack_ca }}"
-  failed_when: false
-  changed_when: false
-  register: logstash_writer_role_present
-  no_log: "{{ elasticstack_no_log }}"
-  run_once: true
-
-# we doubled the task and didn't use a more sophisticated way to just change
-# the URI because we expect this task to be removed when ES 7 is out of
-# support
+- name: Set elasticsearch security-api base url for elasticsearch > 7
+  ansible.builtin.set_fact:
+    security_api_base_url: "https://{{ hostvars[groups['elasticsearch'][0]].ansible_default_ipv4.address }}:{{ elasticstack_elasticsearch_http_port }}/_security/"
```

```suggestion
    security_api_base_url: "https://{{ hostvars[elasticstack_ca].ansible_default_ipv4.address }}:{{ elasticstack_elasticsearch_http_port }}/_security/"
```
Could that work, too? That would make the code a bit more streamlined and easier to maintain.

---

Done ;)

---

Nice. :)